components/nimbus/ios/Nimbus/NimbusApi.swift

```diff
@@ -0,0 +1,141 @@
+     *
+     * This performs the minimum amount of I/O needed to ensure `getExperimentBranch()` is usable.
+     *
+     * It will not take in to consideration previously fetched experiments: `applyPendingExperiments()`
+     * is more suitable for that use case.
+     *
+     * This method uses the single threaded worker scope, so callers can safely sequence calls to
+     * `initialize` and `setExperimentsLocally`, `applyPendingExperiments`.
+     */
+    func initialize()
+
+    /**
+     * Fetches experiments from the RemoteSettings server.
+     *
+     * This is performed on a background thread.
+     *
+     * Notifies `.nimbusExperimentsFetched` to observers once the experiments has been fetched from the
+     * server.
+     *
+     * Notes:
+     * * this does not affect experiment enrolment, until `applyPendingExperiments` is called.
+     * * this will overwrite pending experiments previously fetched with this method, or set with
+     *   `setExperimentsLocally`.
+     */
+    func fetchExperiments()
+
+    /**
+     * Calculates the experiment enrolment from experiments from the last `fetchExperiments` or
+     * `setExperimentsLocally`, and then informs Glean of new experiment enrolment.
+     *
+     * Notifies `.nimbusExperimentsApplied` once enrolments are recalculated.
+     */
+    func applyPendingExperiments()
+
+    /**
+     * Set the experiments as the passed string, just as `fetchExperiments` gets the string from
+     * the server. Like `fetchExperiments`, this requires `applyPendingExperiments` to be called
+     * before enrolments are affected.
+     *
+     * The string should be in the same JSON format that is delivered from the server.
+     *
+     * This is performed on a background thread.
+     */
+    func setExperimentsLocally(_ experimentsJson: String)
+
+    /**
+     * A utility method to load a file from resources and pass it to `setExperimentsLocally(String)`.
+     */
+    func setExperimentsLocally(_ fileURL: URL)
+}
+
+public protocol NimbusUserConfiguration {
+    /**
+     * Opt out of a specific experiment
+     *
+     * - Paramaters
+     *         - experimentId The string experiment-id or "slug" for which to opt out of
+     */
+    func optOut(_ experimentId: String)
+
+    /**
+     * Opt in to a specific experiment with a particular branch.
+     *
+     * For data-science reasons: This should not be utilizable by the the user.
+     */
+    func optIn(_ experimentId: String, branch: String)
+
+    func resetTelemetryIdentifiers(_ identifiers: AvailableRandomizationUnits)
+
+    /**
+     * Control the opt out for all experiments at once. This is likely a user action.
+     */
+    var globalUserParticipation: Bool { get set }
+
+    /**
+     * Get the list of currently enrolled experiments
+     *
+     * @return A list of [EnrolledExperiment]s
+     */
+    func getActiveExperiments() -> [EnrolledExperiment]
```

Comment:
Should we have an API for getting all available experiments?

Reply:
The use-case for `get_all_experiments()` is to build QA tooling to exercise the branches available. It'd be nice if we didn't need to touch this tooling each time we add an experiment.

This is covered in [SDK-228][1].

[1]: https://jira.mozilla.com/browse/SDK-228